src/trades.py

```diff
@@ -0,0 +1,74 @@
+"""
+Definition of trades, orders, and executions.
+"""
+
+from fractions import Fraction
+
+
+class Trade:
+    """
+    Class for executed orders.
+    """
+
+    def __init__(self, data, execution):
+        self.data = data
+        self.execution = execution
+
+    def adapt_execution_to_gas_price(self, gas_price, gas_price_adapted):
+        """
+        Given an executed order which was executed ata time with gas price `gas_price`, adapt the
+        execution to what it would have been with gas price `gas_price_adapted`.
+        """
+        self.execution.adapt_execution_to_gas_price(gas_price, gas_price_adapted)
+
+
+class OrderData:
```

Comment:
This and the one above look like the would be better of declared as dataclasses. Then you would not face the redundancy issues of having to implement an `__init__` method.


Example:


```py
@dataclass
class MyDataClass:
    field_a: int
    field_b: OtherType
    
    
my_instance = MyDataClass(1, instance_of_other_type)
```


vs

```py
class MyRegularClass:

    def __init__(self, field_a: int, field_b: OtherType) -> None:
        self.field_a = field_a
        self.field_b = field_b
```

cf this documentation: https://docs.python.org/3/library/dataclasses.html

Reply:
Is it correct that `dataclass` mostly reduce the "redundant" code for the `__init__`?
I converted all 3 classes into data classes.